.github/workflows/compilation_on_android_ubuntu.yml

```diff
@@ -521,6 +548,10 @@ jobs:
         if: env.USE_LLVM == 'true' && steps.retrieve_llvm_libs.outputs.cache-hit != 'true'
         run: echo "::error::can not get prebuilt llvm libraries" && exit 1
 
+      - name: Build WASI thread tests
```

Comment:
Running the test will download wasi-sdk, build and make wasi-libc, and run build.sh, which may take a lot of time, shall we add the test for some situations, for example, `$DEFAULT_TEST_OPTIONS` test_option for all running modes?

Reply:
Why do we have to build tests? Tests are already compiled and published, so there's no need to rebuild them on CI

Reply:
Can we instead use the wasi-sdk prerelease that already includes threading? https://github.com/WebAssembly/wasi-sdk/releases/tag/wasi-sdk-20%2Bthreads

Reply:
> Running the test will download wasi-sdk, build and make wasi-libc, and run build.sh, which may take a lot of time, shall we add the test for some situations, for example, $WASI_TEST_OPTIONS test_option for all running modes?

Sure, this was just a draft, as you say it's only needed with `$WASI_TEST_OPTIONS`, I'll add the condition.

> Can we instead use the wasi-sdk prerelease that already includes threading? https://github.com/WebAssembly/wasi-sdk/releases/tag/wasi-sdk-20%2Bthreads

Yes, I saw that a few days ago. I was gonna wait for the release, do we want to use the pre-release?

Reply:
I think we could use pre-release and move to release once it's available. Also, if we experience any problems with the pre-release, we can migrate to your current proposal.

Reply:
> I think we could use pre-release and move to release once it's available. Also, if we experience any problems with the pre-release, we can migrate to your current proposal.

Not sure if it makes more sense to do it in a separate PR.

Reply:
I think it would be better to do it in a separate PR, where we would update to `wasi-sdk-20` (pre-release) in all the CI jobs, not only the ones using WASI threads.